crates/workspace/src/pane.rs

```diff
@@ -2615,6 +2629,37 @@ mod tests {
         });
     }
 
+    #[gpui::test]
+    async fn test_add_item_capped_to_max_tabs(cx: &mut TestAppContext) {
```

Comment:
I would also write a test with the `0` setting, since it is a special case. Although _no limit_ is a bit weird to test